### PR TITLE
observability: rename Observability to GcpObservability

### PR DESCRIPTION
**Pull request by sanjaypujare**
rename artifact to grpc-gcp-observability
enable maven publication of grpc-gcp-observability

CC @DNVindhya

### CODE REVIEW

File: gcp-observability/build.gradle [hunk 1]
```diff
@@ -6,7 +6,7 @@ plugins {
     id "ru.vyarus.animalsniffer"
 }
 
-description = "gRPC: Observability"
+description = "gRPC: GcpObservability"
```
Comment (ejona86) [inline gcp-observability/build.gradle:9]: "GCP Observability"? Or "Google Cloud Platform Observability"?

**Review by ejona86 — COMMENTED** [review]
The package and folder structure seem they should also change before enabling the artifact.

**Comment by sanjaypujare** [comment]
> The package and folder structure seem they should also change before enabling the artifact.

Yes, I did think about it but then I thought there is too much repetition of GCP in `io.grpc.gcpobservability.GcpObservability` then decided against it. But I will rename the package to `io.grpc.gcpObservability` (and the corresponding folder structure which follows). The other options are `io.grpc.gcp.observability` or `io.grpc.observability.gcp`. Which one do you think is the best?

**Comment by ejona86** [comment]
Stuttering the package name into the class name is pretty common in Java. Only use lower-case characters though.

When I talked with @dfawley about Go's plans, it sounded like `io.grpc.gcp.observability` would be good for Java and be similar to Go.

**Review by ejona86 — APPROVED** [review]

**Pull request merged**